go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -0,0 +1,1128 @@
+package egressip
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/iprulemanager"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/iptables"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/linkmanager"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/routemanager"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/syncmap"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	eipv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1"
+	egressipinformer "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1/apis/informers/externalversions/egressip/v1"
+	egressiplisters "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1/apis/listers/egressip/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
+
+	"github.com/gaissmai/cidrtree"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	rulePriority        = 6000 // the priority of the ip routing rules created by the controller. Egress Service priority is 5000.
+	routingTableIDStart = 1000
+	chainName           = "OVN-KUBE-EGRESS-IP-MULTI-NIC"
+	iptChainName        = utiliptables.Chain(chainName)
+	maxRetries          = 15
+)
+
+var (
+	_, defaultV4AnyCIDR, _ = net.ParseCIDR("0.0.0.0/0")
+	_, defaultV6AnyCIDR, _ = net.ParseCIDR("0:0:0:0:0:0:0:0")
+	iptJumpRule            = []iptables.RuleArg{{Args: []string{"-j", chainName}}}
+)
+
+// eIP represents exactly one Egress IP IP
+type eIP struct {
+	// EgressIP name
+	name string
+	// EgressIP IP
+	ip        *netlink.Addr
+	routeLink *routemanager.RoutesPerLink
+	// pod specific configuration for ip rules and iptables. Each entry represents one pod IP.
+	podInfos *podIPConfigList
+}
+
+func newEIP() eIP {
+	return eIP{
+		podInfos: newPodIPConfigList(),
+	}
+}
+
+// state contains current state for an EgressIP as it was applied.
+type state struct {
+	// namespaceName -> pod ns/name -> pod IP configuration
+	namespacesWithPodIPConfigs map[string]map[ktypes.NamespacedName]*podIPConfigList
+	// eIP IP contains all applied configuration for a given EgressIP IP
+	eIP
```

Comment:
this can be a bit confusing, since both `namespacesWithPodIPConfigs` and `eIP.podInfos` contain `podIPConfigList`. I see form the update function, that `namespacesWithPodIPConfigs` is actually set as a result of applying the config, but in the cleanup part `existing.eIP.podInfos` are used. So I am confused, as I would expect only one of those to be used to store the status

Reply:
Refactoring to remove dup info